doc(setting): eventual danger zone setting update (#858)
longhorn/website: content/docs/1.6.0/references/settings.md

> **Warning:**
>  - Specifying a value of 0 disables CPU requests for instance manager pods. You must specify an integer between 1000 and 8000. 

> >  - This is a global setting. Modifying the value triggers an automatic restart of the Instance Manager pods. However, V2 Instance Manager pods that use this setting are restarted only when no instances are running.

Reviewer: Why only v2?

Author: Because this is the section for `Guaranteed Instance Manager CPU for V2 Data Engine`
Here is for V1 https://github.com/longhorn/website/pull/858/files#diff-fa0cc835e2d54b4934afb441d3f5124d01daa7556cf3d1c2037ee1bbd4462d06R876

> >  - This is a global setting. Modifying the value triggers an automatic restart of the instance manager pods. After this setting is changed, the v2 instance manager pod using this global setting will be automatically restarted without instances running on the v2 instance manager.

Reviewer 2: > v2 instance manager pod using this global setting will be automatically restarted without instances running on the v2 instance manager.

This is unclear. Do you mean that the pod is restarted only when no instances are running on the Instance Manager?

Author: Yes, it is.

### Danger Zone

> Longhorn allows you to promptly update the Danger Zone settings ([Kubernetes Taint Toleration](#kubernetes-taint-toleration), [Priority Class](#priority-class), [System Managed Components Node Selector](#system-managed-components-node-selector), [Storage Network](#storage-network), and [Guaranteed Instance Manager CPU](#guaranteed-instance-manager-cpu)) without the need to wait for all volumes detached. It is designed to provide users with more flexibility and efficiency in managing their configurations. To achieve the immediate effect, one of the following conditions must be met:
> 
> - No Attached Volumes: If there are no volumes attached, the setting change will take effect immediately.
> - Engine Image Upgrade (Volume Live Upgrade): During a volume live upgrade, which involves creating a new instance manager pod, the setting change will take effect immediately for the new pod.
> 
> If the immediate effect conditions are not met, the setting change is reconciled hourly, taking effect when all volumes are detached.
> 
> When all volumes are detached, the settings in the following table are immediately applied and the system-managed components (e.g, instance manager, engine image, CSI driver, etc.) are restarted. If you do not detach all volumes before the settings are synchronized, the settings are not applied immediately as mentioned above and you might need to reconfigure the settings after detaching the remaining volumes; otherwise, you can wait for the setting change to be reconciled in an hour.

Reviewer 2: ```suggestion
Starting with Longhorn v1.6.0, Longhorn allows you to modify the [Danger Zone settings](https://longhorn.io/docs/1.6.0/references/settings/#danger-zone) without the need to wait for all volumes to become detached. Your preferred settings are immediately applied in the following scenarios:

- No attached volumes: When no volumes are attached before the settings are configured, the setting changes are immediately applied.
- Engine image upgrade (live upgrade): During a live upgrade, which involves creating a new Instance Manager pod, the setting changes are immediately applied to the new pod.

Settings are synchronized hourly. When all volumes are detached, the settings in the following table are immediately applied and the system-managed components (for example, Instance Manager, CSI Driver, and engine images) are restarted. If you do not detach all volumes before the settings are synchronized, the settings are not applied and you must reconfigure the same settings after detaching the remaining volumes.
```

Author: Updated. Thanks

You need to set node selector for both of them. This setting only sets node selector for system managed components. Follow the instruction at [Node Selector](../../advanced-resources/deploy/node-selector) to change node selector.

> **Warning:**  Since all Longhorn components will be restarted, the Longhorn system is unavailable temporarily.

> Make sure all Longhorn volumes are `detached`. If there are running Longhorn volumes in the system, this means the Longhorn system cannot restart its components and you might need to set this setting again when all Longhorn volumes are `detached`.

Reviewer 2: ```suggestion
Ensure that all Longhorn volumes are detached. When volumes are in use, Longhorn components are not restarted so you might be required to configure this setting again when all volumes are finally detached.
```

Author: Modified. Thanks.

The storage network uses Multus NetworkAttachmentDefinition to segregate the in-cluster data traffic from the default Kubernetes cluster network.

> > **Warning:** This setting should change after detaching all Longhorn volumes, as some of the Longhorn system component pods will get recreated to apply the setting. When applying the setting, Longhorn will try to restart all instance-manager, and backing-image-manager pods if all volumes are detached and eventually restart the instance manager pod without instances running on the instance manager.

Reviewer 2: ```suggestion
> **Warning:** This setting should change after all Longhorn volumes are detached because some pods that run Longhorn system components are recreated to apply the setting. When all volumes are detached, Longhorn attempts to restart all Instance Manager and Backing Image Manager pods.
```

Reviewer 2: > eventually restart the instance manager pod without instances running on the instance manager.

This is unclear. Do you mean that the pod is restarted only when no instances are running on the Instance Manager?

Author: Yes, just like what you mentioned.

Author: Modified. Thanks.

You need to set node selector for both of them. This setting only sets node selector for system managed components. Follow the instruction at [Node Selector](../../advanced-resources/deploy/node-selector) to change node selector.

> **Warning:**  Since all Longhorn components will be restarted, the Longhorn system is unavailable temporarily.

> Ensure that all Longhorn volumes are detached. When volumes are in use, Longhorn components are not restarted so you might be required to configure this setting again when all volumes are finally detached.

Reviewer: Why need to ensure all volumes are detached? Also, don't use `might be required`, as this would be unclear for users. Whether they need to reconfigure or not, this should be very clear. 

You can briefly explain the function and when it will be taken. For the details, refer to the advanced resource page. 

The same suggestion for other danger zone settings in this setting page.

Author: Updated to remove `might` and add a brief explanation.

>  - Considering the possible number of new instance manager pods in a further system upgrade, this integer value ranges from 0 to 40.
>  - One more set of instance manager pods may need to be deployed when the Longhorn system is upgraded. If current available CPUs of the nodes are not enough for the new instance manager pods, you need to detach the volumes using the oldest instance manager pods so that Longhorn can clean up the old pods automatically and release the CPU resources. And the new pods with the latest instance manager image will be launched then.
>  - This global setting will be ignored for a node if the field "InstanceManagerCPURequest" on the node is set.

> >  - After this setting is changed, the v1 instance manager pod using this global setting will be automatically restarted without instances running on the v1 instance manager.

Reviewer 2: I'll review this line after "without instances running on the v1 instance manager" is clarified.

Author: > "without instances running on the v1 instance manager"

Yes, like what you mentioned above.
The instance manager pod will be restarted only when no instances are running on the Instance Manager and the setting is not applied to the pod.

Reviewer 2: ```suggestion
>  - After the setting is changed, the V2 Instance Manager pods that use this setting are automatically restarted when no instances are running.
```

Author: Modified. Thanks.
